go-controller/pkg/ovn/egressip.go

```diff
@@ -2099,12 +2100,18 @@ func (oc *Controller) checkEgressNodesReachability() {
 }
 
 func (oc *Controller) isReachable(node *egressNode) bool {
-	for _, ip := range node.mgmtIPs {
-		if dialer.dial(ip) {
-			return true
+	if err := wait.PollImmediate(5*time.Second, 2*time.Minute, func() (bool, error) {
```

Comment:
A 2 minute timeout is quite risky. If an egress node really has a problem and is unreachable, then this timeout will hold all calls to `isReachable` for a very long time. Importantly: it will make the go routine which calls `checkEgressNodesReachability` very slow if there is a problematic node, and make it react to node changes very slowly. I think this code needs to do something similar to openshift-sdn, i.e: retry 2/3 times in quick succession and then just flag the node not reachable if the dials fails during those calls.

Reply:
I agree with this, it's quite intense to go from a 1s timeout with 5s intervals straight to a 2-minute timeout
If this is based on the discussions from the BZ2083425, then a 5s timeout may be suitable as this is what's done in OpenShift SDN:

Retries In OpenShift SDN : 
- https://github.com/openshift/sdn/blob/master/pkg/network/node/egressip.go/#L247-L259
- https://github.com/openshift/sdn/blob/master/pkg/network/node/egressip.go/#L208-L214

Timeout (default) in OpenShift SDN is the full 5 seconds not 1 second:
- https://github.com/openshift/sdn/blob/master/pkg/network/node/egressip.go/#L24-L28
- https://github.com/openshift/sdn/blob/master/pkg/network/node/egressip.go/#L228-L243
- https://github.com/openshift/sdn/blob/master/pkg/network/common/egressip.go/#L715-L717

Reply:
I agree with this, it's quite intense to go from a 1s single failure threshold directly to a 2-minute failure threshold.

When there's a real EgressIP Node failure, this is going to cause a really long outage.

There are discussions in BZ2083425 about a similar thing, 5s timeout may be suitable with shorter retries, as is done with OpenShift SDN:

Retries: 
- https://github.com/openshift/sdn/blob/master/pkg/network/node/egressip.go/#L247-L259
- https://github.com/openshift/sdn/blob/master/pkg/network/node/egressip.go/#L208-L214

Scaling Timeout:
- https://github.com/openshift/sdn/blob/master/pkg/network/node/egressip.go/#L24-L28
- https://github.com/openshift/sdn/blob/master/pkg/network/node/egressip.go/#L228-L243
- https://github.com/openshift/sdn/blob/master/pkg/network/common/egressip.go/#L715-L717